### DISCUSSION TITLE
How to use TextLayerBuilder to render text on pdf?

### DISCUSSION THREAD

**Discussion by AscEzio** [mozilla/pdf.js#17607] [Q&A]
Attach (recommended) or Link to PDF file here: No(Because of developing)

My problem: I've been searching online for how to use TextLayerBuilder to render text onto the DOM, but the text isn't being rendered to the specified DOM element; the target DOM is empty. What's more, I have many doubt about TextLayerBuilder.
1. The paramters I learned online is different from the type document of TextLayerBuilder(In typescript).
![image](https://github.com/mozilla/pdf.js/assets/92004703/20077e9a-7372-4272-9aba-bca9883a1c79)
2. Many Online documents provider some non-existent API, like setTextContent(it should be setTextContentSource), enableTextSelection.

References:
1. https://blog.51cto.com/u_11022239/3156166
2. https://www.volcengine.com/theme/4376967-R-7-1
3. https://www.jianshu.com/p/be452bb306a5

Configuration:
- Web browser and its version: chrome, version 121.0.6167.85
- Operating system and its version: windows 10, version 19045.3930
- PDF.js version: 3.11.174
- Is a browser extension: No

Steps to reproduce the problem:

What is the expected behavior? (add screenshot)
I hope it can be like this. Users can select text and copy.
![image](https://github.com/mozilla/pdf.js/assets/92004703/aaebe00a-5536-42c4-b40f-1cb73f2a03ed)

What went wrong? (add screenshot)
The dom that render text is empty.
![image](https://github.com/mozilla/pdf.js/assets/92004703/e454b2a7-1719-4232-947d-08a56c8a6b1b)

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):
Here is my project in github:
https://github.com/AscEzio/pdfViewer/blob/fix/TextLayerBuilder/src/utils/renderPDF.js

Sorry for my poor english and front-end technique

**Comment by lfoppiano** [comment]
@AscEzio could you share the code here? I'm investigating the same problem, and I've been facing the same issues as you did.
